packages/artifact/src/internal/download/download-artifact.ts

```diff
@@ -99,46 +89,8 @@ export async function streamExtractExternal(
         clearTimeout(timer)
         reject(error)
       })
-      .pipe(unzip.Parse())
-      .pipe(
-        new stream.Transform({
-          objectMode: true,
-          transform: async (entry, _, callback) => {
-            const fullPath = path.normalize(path.join(directory, entry.path))
-            if (!directory.endsWith(path.sep)) {
-              directory += path.sep
-            }
-            if (!fullPath.startsWith(directory)) {
-              reject(new Error(`Malformed extraction path: ${fullPath}`))
-            }
-
-            if (entry.type === 'Directory') {
-              if (!createdDirectories.has(fullPath)) {
-                createdDirectories.add(fullPath)
-                await resolveOrCreateDirectory(fullPath).then(() => {
-                  entry.autodrain()
-                  callback()
-                })
-              } else {
-                entry.autodrain()
-                callback()
-              }
-            } else {
-              core.info(`Extracting artifact entry: ${fullPath}`)
-              if (!createdDirectories.has(path.dirname(fullPath))) {
-                createdDirectories.add(path.dirname(fullPath))
-                await resolveOrCreateDirectory(path.dirname(fullPath))
-              }
-
-              const writeStream = createWriteStream(fullPath)
-              writeStream.on('finish', callback)
-              writeStream.on('error', reject)
-              entry.pipe(writeStream)
-            }
-          }
-        })
-      )
-      .on('finish', async () => {
+      .pipe(unzip.Extract({path: directory}))
```

Comment:
Glad we don't have to roll our own implementation anymore

Reply:
Agreed! Simplifies this a ton.